service/src/main/java/bio/terra/externalcreds/services/ProviderService.java

```diff
@@ -164,6 +169,64 @@ public void deleteLink(String userId, String providerId) {
     linkedAccountService.deleteLinkedAccount(userId, providerId);
   }
 
+  @VisibleForTesting
+  void authAndRefreshPassport(LinkedAccount linkedAccount) {
+    if (linkedAccount.getExpires().before(Timestamp.from(Instant.now()))) {
+      var linkedAccountId =
+          linkedAccount
+              .getId()
+              .orElseThrow(() -> new ExternalCredsException("linked account id missing"));
+      passportService.deletePassport(linkedAccountId);
+      linkedAccountService.upsertLinkedAccount(linkedAccount.withIsAuthenticated(false));
```

Comment:
```suggestion
      linkedAccountService.upsertLinkedAccountWithPassportAndVisas(
          new LinkedAccountWithPassportAndVisas.Builder()
              .linkedAccount(linkedAccount.withIsAuthenticated(false))
              .build());
```
this is nice because the single call is transactional